src/test/java/com/puppycrawl/tools/checkstyle/AbstractTreeTestSupport.java

```diff
@@ -0,0 +1,116 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public abstract class AbstractTreeTestSupport extends AbstractPathTestSupport {
+
+    protected static final String LF_REGEX = "\\\\n";
+
+    protected static final String CLRF_REGEX = "\\\\r\\\\n";
+
+    /**
+     * Returns canonical path for the file with the given file name.
+     * The path is formed base on the non-compilable resources location.
+     * This implementation uses 'src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/'
+     * as a non-compilable resource location.
+     * @param filename file name.
+     * @return canonical path for the file with the given file name.
+     * @throws IOException if I/O exception occurs while forming the path.
+     */
+    protected String getNonCompilablePath(String filename) throws IOException {
+        return new File("src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/"
```

Comment:
@rnveach 
I don't think I can use `getPackageLocation()` because it overrides the method in the super class and has the path `com/puppycrawl/tools/checkstyle/astprinter/` but this path does not exist in `resources-noncompilable`. There is no `astprinter` folder